cln/cln.go

```diff
@@ -192,18 +193,18 @@ func (c *Cln) CreateInvoice(value int64, preimage []byte, expiry int64, memo str
 	}, nil
 }
 
-func (c *Cln) PayInvoice(invoice string, feeLimit uint, timeoutSeconds uint, chanId lightning.ChanId) (*lightning.PayInvoiceResponse, error) {
+func (c *Cln) PayInvoice(invoice string, feeLimit uint, timeoutSeconds uint, chanIds []lightning.ChanId) (*lightning.PayInvoiceResponse, error) {
 	retry := uint32(timeoutSeconds)
 
 	var exclude []string
 
-	if chanId != 0 {
+	if len(chanIds) > 0 && chanIds[0] != 0 {
```

Comment:
```suggestion
	if len(chanIds) > 0 {
```
Why do we still need this check?

Reply:
Autoswap recommendations include a channel id, which is passed to the create swap call, but the zero value means no channel. I guess that check could be put somewhere else